lib/particles/moving_particle.dart

```diff
@@ -30,8 +29,10 @@ class MovingParticle extends CurvedParticle with SingleChildParticle {
   @override
   void render(Canvas c) {
     c.save();
-    final Offset current = Offset.lerp(from, to, progress);
-    c.translate(current.dx, current.dy);
+    final Offset? current = Offset.lerp(from, to, progress);
```

Comment:
how could this return null?

Reply:
This one is interesting. If both `from` and `to` are `null` it returns `null`. However. we use non-null values for both. So, in our case, it never returns null.

Forcing non-null here then.